gpcc-mocks/src/main/java/uk/nhs/adaptors/gpccmocks/gpc/GpcController.java

```diff
@@ -0,0 +1,208 @@
+        @RequestHeader(name = "Ssp-To") String sspTo,
+        @RequestHeader(name = "Ssp-InteractionID") String sspInteractionId,
+        @RequestHeader(name = HttpHeaders.AUTHORIZATION) String authorization,
+        @RequestBody String requestBody) {
+
+        var host = getHostAndPortFromRequest(request);
+
+        log.debug("Request for 'Access Structured Record'. " +
+                "odsCode={} Ssp-TraceID={} Ssp-From={} Ssp-To={} Ssp-InteractionID={} Host/X-Forwarded-Host: {}",
+            odsCode, sspTraceId, sspFrom, sspTo, sspInteractionId, host);
+
+        log.debug("Request body for 'Access Structured Record'\n{}", requestBody);
+
+        if (!isUuid(sspTraceId)) {
+            return badRequest("Ssp-TraceID header must be a UUID");
+        }
+
+        var result = (List<?>) JsonPath.read(requestBody, "$.parameter[:1].valueIdentifier.value");
+        if (result.isEmpty()) {
+            return badRequest("Request body does not include patient identifier");
+        }
+        var nhsNumber = (String) result.get(0);
+
+        GpcModel.GpcModelBuilder gpcModelBuilder = GpcModel.builder()
+            .baseUrl("http://" + host)
+            .odsCode(odsCode);
+
+        switch (nhsNumber) {
+            case "9690937294":
+            case "9690937286":
+                gpcModelBuilder.nhsNumber(nhsNumber);
+                break;
+            default:
+                return patientNotFound("Patient " + nhsNumber + " not found");
+        }
+
+        var body = TemplateUtils.fillTemplate("gpc/accessRecordStructured", gpcModelBuilder.build());
+        return new ResponseEntity<>(body, getResponseHeaders(), HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/{odsCode}/STU3/1/gpconnect/documents/fhir/Patient/{patientId}/DocumentReference")
+    @ResponseStatus(value = ACCEPTED)
+    public ResponseEntity<String> findPatientsDocuments(
+        HttpServletRequest request,
+        @PathVariable String odsCode,
+        @PathVariable String patientId,
+        @RequestHeader(name = "Ssp-TraceID") String sspTraceId,
+        @RequestHeader(name = "Ssp-From") String sspFrom,
+        @RequestHeader(name = "Ssp-To") String sspTo,
+        @RequestHeader(name = "Ssp-InteractionID") String sspInteractionId,
+        @RequestHeader(name = HttpHeaders.AUTHORIZATION) String authorization) {
+
+        var host = getHostAndPortFromRequest(request);
+
+        log.debug("Request for 'Find a patient's documents'. " +
+                "odsCode={} Ssp-TraceID={} Ssp-From={} Ssp-To={} Ssp-InteractionID={} Host/X-Forwarded-Host: {}",
+            odsCode, sspTraceId, sspFrom, sspTo, sspInteractionId, host);
+
+        if (!isUuid(sspTraceId)) {
+            return badRequest("Ssp-TraceID header must be a UUID");
+        }
+
+        GpcModel.GpcModelBuilder gpcModelBuilder = GpcModel.builder()
+            .baseUrl("http://" + host)
+            .odsCode(odsCode)
+            .patientId(patientId);
+
+        switch (patientId) {
```

Comment:
the patient in the response bundle always has id = 2, even if it was 1 requested.

Reply:
The GPCC and GP2GP adaptors don't use the Patient resource included in the "Find a patient's documents" response. I don't think this has an impact on any further development work. Could you create a backlog ticket for the improvement?

P.S. There are lots of other references to FHIR resources in these mocks which are invalid

Reply:
I know, but it looks like quick win when there is already prepared partials/patient